airflow-core/src/airflow/api_fastapi/core_api/datamodels/tasks.py

```
@@ -93,11 +96,11 @@ def validate_weight_rule(cls, wr: str | PriorityWeightStrategy | None) -> str |
 
     @field_validator("params", mode="before")
     @classmethod
-    def get_params(cls, params: abc.MutableMapping | None) -> dict | None:
+    def get_params(cls, params: SerializedParamsDict | None) -> dict | None:
         """Convert params attribute to dict representation."""
         if params is None:
             return None
-        return {param_name: param_val.dump() for param_name, param_val in params.items()}
+        return {k: v.resolve(suppress_exception=True) for k, v in params.items()}
```

Would we ever need to suppress exception here though? Wouldn't it be captured before serialization anyway?

---

It’s still possible for this to raise if no default is provided. I guess it would be possible to make this _never_ raise (returning None without default) instead, but it’s also somewhat awkward to have a different interface to the sdk class.